Sources/BraveWallet/Panels/Signature Request/SignInWithEthereumView.swift

```diff
@@ -0,0 +1,243 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveStrings
+import BraveCore
+import DesignSystem
+
+/// View for showing `SignMessageRequest` for  ethSiweData
+struct SignInWithEthereumView: View {
+  
+  let account: BraveWallet.AccountInfo
+  let originInfo: BraveWallet.OriginInfo
+  let message: BraveWallet.SIWEMessage
+  var action: (_ approved: Bool) -> Void
+  
+  @State private var isShowingDetails: Bool = false
+  @Environment(\.sizeCategory) private var sizeCategory
+  
+  var body: some View {
+    ScrollView {
+      VStack(spacing: 10) {
+        faviconAndOrigin
+        
+        messageContainer
+        
+        buttonsContainer
+          .padding(.top)
+          .opacity(sizeCategory.isAccessibilityCategory ? 0 : 1)
+          .accessibility(hidden: sizeCategory.isAccessibilityCategory)
+      }
+      .padding()
+    }
+    .overlay(
+      Group {
+        if sizeCategory.isAccessibilityCategory {
+          buttonsContainer
+            .frame(maxWidth: .infinity)
+            .padding(.top)
+            .background(
+              LinearGradient(
+                stops: [
+                  .init(color: Color(.braveGroupedBackground).opacity(0), location: 0),
+                  .init(color: Color(.braveGroupedBackground).opacity(1), location: 0.05),
+                  .init(color: Color(.braveGroupedBackground).opacity(1), location: 1),
+                ],
+                startPoint: .top,
+                endPoint: .bottom
+              )
+              .ignoresSafeArea()
+              .allowsHitTesting(false)
+            )
+        }
+      },
+      alignment: .bottom
+    )
+    .background(Color(braveSystemName: .containerHighlight))
+    .navigationTitle(Strings.Wallet.signInWithBraveWallet)
+  }
+  
+  private var faviconAndOrigin: some View {
+    VStack(spacing: 8) {
+      OriginInfoFavicon(originInfo: originInfo)
+      Text(verbatim: originInfo.eTldPlusOne)
+      Text(originInfo: originInfo)
+        .font(.caption)
+        .foregroundColor(Color(.braveLabel))
+        .multilineTextAlignment(.center)
+    }
+  }
+  
+  private var messageContainer: some View {
+    VStack(alignment: .leading, spacing: 10) {
+      AddressView(address: account.address) {
+        AccountView(
+          address: account.address,
+          name: account.name,
+          blockieShape: .rectangle
```

Comment:
Are we married to this square icon? It kinda looks weird lol, could we at least soften it up with some rounded corners?

Reply:
This was setup this way on `Blockie` with a regular rectangle so the parent view could apply it's own corner radius / style. I think we can either use a scaled metric in `Blockie`, or apply rounded rectangle clipping to the blockie in `AccountView` 👍